atc/api/auth/check_authorization_handler.go

```diff
@@ -30,6 +31,13 @@ func (h checkAuthorizationHandler) ServeHTTP(w http.ResponseWriter, r *http.Requ
 	}
 
 	teamName := r.URL.Query().Get(":team_name")
+	if teamName == "" {
+		pipeline, ok := r.Context().Value(PipelineContextKey).(db.Pipeline)
+		if !ok {
+			panic("missing pipeline")
+		}
+		teamName = pipeline.TeamName()
+	}
```

Comment:
something feels fishy to me about this -- it seems like this code is a bit obsessed with the implementation details of the fetchpipelinewrappa. It's not a huge deal, but aesthetically it feels funny. Would it make sense to have a fetchteamwrappa so that there's a uniform way to get the team from a requestcontext as well?

Reply:
It is definitely fishy, but I was struggling to come up with a better solution.

One approach I had was to modify the `check_pipeline_access_handler` to support the case where we allow/disallow public pipelines, and use that for these endpoints instead, which is probably a slight improvement - although, I do the same awkward thing in `policychecker/checker.go`, which currently runs for every endpoint.

As you pointed out, I think the bigger problem is the implicit passing of data through the request context, since we lose all type safety. I'll try to think of alternative approaches, but definitely let me know if you can come up with anything